src/pages/Account/Tabs.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from "react";
+import Tabs from "@mui/material/Tabs";
+import Tab from "@mui/material/Tab";
+import Typography from "@mui/material/Typography";
+import Box from "@mui/material/Box";
+import RawDataTab from "./RawDataTab/Index";
+import TransactionTab from "./TransactionsTab/Index";
+
+function tabProps(index: number) {
+  return {
+    "aria-controls": `tabpanel-${index}`,
+  };
+}
+
+type TabPanelProps = {
+  children?: React.ReactNode;
+  index: number;
+  value: number;
+};
+
+function TabPanel(props: TabPanelProps): JSX.Element {
+  const {children, value, index} = props;
+
+  return (
+    <div hidden={value !== index} id={`tabpanel-${index}`}>
+      {value === index && (
+        <Box sx={{p: 3}}>
+          <Typography>{children}</Typography>
+        </Box>
+      )}
+    </div>
+  );
+}
+
+type AccountTabsProps = {
+  address: string;
+};
+
+export default function AccountTabs({address}: AccountTabsProps): JSX.Element {
+  const [tabIdx, setTabIdx] = React.useState(0);
+
+  const handleChange = (event: React.SyntheticEvent, newIdx: number) => {
+    setTabIdx(newIdx);
+  };
+
+  // TODO: use LinkTab for better navigation
+  return (
+    <Box sx={{width: "100%"}}>
+      <Box sx={{borderBottom: 1, borderColor: "divider"}}>
+        <Tabs
+          value={tabIdx}
+          onChange={handleChange}
+          aria-label="basic tabs example"
+        >
+          <Tab
+            label="Account Transactions"
+            sx={{fontSize: "large"}}
+            {...tabProps(0)}
```

Comment:
I think it makes sense to create a data structure with the list of tabs (labels + components), then render based on that data structure (instead of hardcoding it like this).

Reply:
I added the map for tab panel components. However, mapping the tab components does not work. It seems like `onChange` only works well when the `<Tab />` components are directly inside `<Tabs />`. I'm hoping that we can skip it here and get back to this issue later. 

The code I tried that didn't work: 
```import * as React from "react";
import {useState} from "react";
import {Tabs, Tab, Box} from "@mui/material";
import RawDataTab from "./RawDataTab/Index";
import TransactionTab from "./TransactionsTab/Index";
import {assertNever} from "../../utils";

const TabComponents = Object.freeze({
  transactions: TransactionTab,
  rawData: RawDataTab,
});

type TabValue = keyof typeof TabComponents;

const TAB_VALUES: TabValue[] = ["transactions", "rawData"];

type TabPanelProps = {
  value: TabValue;
  address: string;
};

function TabPanel({value, address}: TabPanelProps): JSX.Element {
  switch (value) {
    case "transactions":
      return <TransactionTab address={address} />;
    case "rawData":
      return <RawDataTab address={address} />;
    default:
      return assertNever(value);
  }
}

function TabLabel({value}: {value: TabValue}) {
  switch (value) {
    case "transactions":
      return (
        <Tab
          label="Account Transactions"
          value="transactions"
          sx={{fontSize: "large"}}
        />
      );
    case "rawData":
      return <Tab label="Raw Data" value="rawData" sx={{fontSize: "large"}} />;
    default:
      return assertNever(value);
  }
}

type AccountTabsProps = {
  address: string;
};

export default function AccountTabs({address}: AccountTabsProps): JSX.Element {
  const [value, setValue] = useState<TabValue>(TAB_VALUES[0]);

  const handleChange = (event: React.SyntheticEvent, newValue: TabValue) => {
    setValue(newValue);
  };

  // TODO: use LinkTab for better navigation
  return (
    <Box sx={{width: "100%"}}>
      <Box sx={{borderBottom: 1, borderColor: "divider"}}>
        <Tabs
          value={value}
          onChange={handleChange}
          aria-label="account page tabs"
        >
          {TAB_VALUES.map((value) => (
            <TabLabel key={value} value={value} />
          ))}
        </Tabs>
      </Box>
      <Box sx={{marginY: 3}}>
        <TabPanel value={value} address={address} />
      </Box>
    </Box>
  );
}
```

Reply:
You need to take in `{...rest}` and apply it to the Tab component:

```jsx
function TabLabel({value, ...rest}: {value: TabValue}) {
  switch (value) {
    case "transactions":
      return (
        <Tab
          label="Account Transactions"
          value="transactions"
          sx={{fontSize: "large"}}
          {...rest}
        />
      );
    case "rawData":
      return (
        <Tab
          label="Raw Data"
          value="rawData"
          sx={{fontSize: "large"}}
          {...rest}
        />
      );
    default:
      return assertNever(value);
  }
}
```

This is because the `TabPanel` component adds additional props to its children (e.g. onClick, without which your handleChange function won't be called).

Reply:
Thank you so much for helping debugging this!!! Let me try it out.